packages/vitest-pool-workers/src/config/index.ts

```diff
@@ -161,17 +144,35 @@ function ensureWorkersConfig<T extends UserConfig>(config: T): T {
 	return config;
 }
 
+export function defineWorkersConfig(
+	config: WorkersUserConfigExport
+): WorkersUserConfigExport;
+export function defineWorkersConfig(
+	config: Promise<WorkersUserConfigExport>
+): Promise<WorkersUserConfigExport>;
+export function defineWorkersConfig(
+	config: ConfigFn<WorkersUserConfigExport>
+): ConfigFn<WorkersUserConfigExport>;
 export function defineWorkersConfig(
 	config: AnyConfigExport<WorkersUserConfigExport>
 ): AnyConfigExport<WorkersUserConfigExport> {
-	return mapAnyConfigExport(ensureWorkersConfig, config);
+	if (typeof config === "function") {
```

Comment:
I merged `defineWorkersConfig` and `mapAnyConfigExport`. Otherwise,  I would have to write overload signatures  to both methods